plugins/inputs/docker/docker.go

```diff
@@ -322,6 +337,27 @@ func (d *Docker) gatherContainer(
 		}
 	}
 
+	// Add whitelisted environment variables to tags
+	if len(d.GatherEnvironment) > 0 {
+		info, err := inspectWrapper(d.client, ctx, container.ID)
+		if err != nil {
+			return fmt.Errorf("Error inspecting docker container: %s", err.Error())
+		}
+		for _, envvar := range info.Config.Env {
+			for _, prefix := range d.GatherEnvironment {
+				if strings.Count(envvar, "=") > 1 {
```

Comment:
I think we should't skip environment variables like: `FOO=BAR=BAZ`, also add a testcase for this.